odoo/modules/loading.py

```diff
@@ -551,8 +551,12 @@ def load_modules(
         #   - module C is loaded and extends model M;
         #   - module B and C depend on A but not on each other;
         # The changes introduced by module C are not taken into account by the upgrade of B.
+        if update_module:
+            # We need to fix custom fields for which we have dropped the not-null constraint.
+            cr.execute("""SELECT DISTINCT model FROM ir_model_fields WHERE state = 'manual'""")
+            models_to_check.update(model_name for model_name, in cr.fetchall() if model_name in registry)
         if models_to_check:
```

Comment:
Another thought of something:
Maybe I can make another PR after this one
1. this code will be skipped if there are both `to upgrade` and `to remove` modules
2. `init_models` should work together with `finalize_constraints`
maybe we should move this and `# STEP 3.6: apply remaining constraints in case of an upgrade` into `# STEP 4: Finish and cleanup installations` 🤔

Reply:
You may, but as you're saying, in another PR. :)